_posts/2019-02-13-extending-pg-search-sort.md

```diff
@@ -0,0 +1,100 @@
+---
+layout: post
+logo: "--white"
+header_style: "c-header--black"
+title: "Extending sort capabilities of pg_search"
+date: "Feb 13, 2019"
+author: "Roel Bondoc"
+author_role: "Fullstack Developer Razeware"
+author_bio: "Interests: Ruby on Rails, games, and basketball."
+author_image: "roel-bondoc@2x.jpeg"
+color: "#ffcb32"
+hero: "c-post-hero--dark"
+image: "/assets/img/pattern-6@2x.png"
+category: "development"
+excerpt: "Adjusting the sort capabilities pg_search to provide better search results."
+---
+
+The search functionality of raywenderlich.com is powered by Postgresql using the `pg_search` ruby gem. Out of the box it is a fantastic library and makes building a search feature easy. Although, as easy as it was, we were quick to discover that our search results were not returning desired results. So I decided to take a closer look at how `pg_search` constructs its queries.
+
+## The default sort
+By default `pg_search` sorts results by a rank result of all the queried columns combined as a single value. In addition, you can set weights on each column so that certain columns will return a higher value than others. The configuration looks something like this:
```

Comment:
What does this mean? A little more background on the default ranking system (doesn't have to be technical) will add the context needed to understand the problem later on.